src/test/java/fi/otavanopisto/kuntaapi/test/AbstractPtvMocker.java

```diff
@@ -0,0 +1,493 @@
+        if (!isMocked(id)) {
+          mockEntity(id, readEntity(id));
+        } else {
+          setStatus(id, MockedResourceStatus.OK);
+        }
+      } catch (JsonProcessingException e) {
+        System.err.println(String.format("Failed to read %s of %s", id, getName()));
+        fail(e.getMessage());
+      }
+    }
+    
+    return this;
+  }
+
+  public AbstractPtvMocker<R> unmock(String... ids) {
+    for (String id : ids) {
+      setStatus(id, MockedResourceStatus.NOT_FOUND);
+    }
+    
+    return this;
+  }
+  
+  protected void mockEntity(String id, R entity) throws JsonProcessingException {
+    add(id, entity, urlPathEqualTo(String.format("%s/%s", getBasePath(), id)));
+  }
+  
+  private void mockDefaultLists() {
+    Map<String, StringValuePattern> queryParams = new HashMap<>();
+    queryParams.put("page", containing("0"));
+    
+    addStatusList(MockedResourceStatus.OK, urlPathEqualTo(getBasePath()));
+    addStatusList(MockedResourceStatus.OK, urlPathEqualTo(getBasePath()), queryParams);
+  }
+  
+  private void updateSubMockerStatuses(String id, MockedResourceStatus status) {
+    if (subMockers.containsKey(id)) {
+      for (AbstractPtvMocker<?> subMocker : subMockers.get(id)) {
+        if (status == MockedResourceStatus.OK) {
+          subMocker.start();
+        } else {
+          subMocker.stop();
+        }
+      }
+    }
+  }
+  
+  public void addStatusList(MockedResourceStatus status, UrlPattern urlPattern) {
+    addStatusList(status, urlPattern, null);
+  }  
+
+  public void addStatusList(MockedResourceStatus status, UrlPattern urlPattern, Map<String, StringValuePattern> queryParams) {
+    MappingBuilder mapping = get(urlPattern);
+    
+    if (queryParams != null) {
+      for (Entry<String, StringValuePattern> queryParam : queryParams.entrySet()) {
+        mapping.withQueryParam(queryParam.getKey(), queryParam.getValue());
+      }
+    }
+    
+    if (!statusLists.containsKey(status)) {
+      statusLists.put(status, new ArrayList<>());
+    }
+    
+    statusLists.get(status).add(mapping);
+  }
+
+  public void addSubMocker(String id, AbstractPtvMocker<?> subMocker) {
+    if (!subMockers.containsKey(id)) {
+      subMockers.put(id, new ArrayList<>());
+    }
+    
+    subMockers.get(id).add(subMocker);
+  }
+  
+  protected R readEntity(String id) {
+    return readEntityFromJSONFile(String.format("ptv/%s/%s.json", getName(), id));
+  }
+
+  @SuppressWarnings("unchecked")
+  private R readEntityFromJSONFile(String file) {
+    return readJSONFile(file, getGenericTypeClass());
+  }
+  
+  private R readJSONFile(String file, Class <R> type){
+    ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.registerModule(new JavaTimeModule());
+    
+    try (InputStream stream = getClass().getClassLoader().getResourceAsStream(file)) {
+      return objectMapper.readValue(stream, type);
+    } catch (IOException e) {
+      // logger.log(Level.SEVERE, FAILED_TO_READ_MOCK_FILE, e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)